pkgs/development/python-modules/qpsolvers/default.nix

```
@@ -24,28 +30,38 @@ buildPythonPackage rec {
     hash = "sha256-/HLc9dFf9F/6W7ux2Fj2yJuV/xCVeGyO6MblddwIGdM=";
   };
 
-  pythonImportsCheck = [ "qpsolvers" ];
+  nativeBuildInputs = [
+    flit-core
+  ];
 
   propagatedBuildInputs = [
-    daqp
-    ecos
     numpy
-    osqp
     scipy
-    scs
   ];
 
   nativeCheckInputs = [
-    flit-core
-    quadprog
     unittestCheckHook
+  ] ++ passthru.optional-dependencies.solvers;
+
+  pythonImportsCheck = [
+    "qpsolvers"
+  ];
+
+  passthru.optional-dependencies.solvers = [
+    clarabel
+    cvxopt
+    daqp
+    ecos
+    osqp
+    quadprog
+    scs
   ];
```

I think add `passthru.optional-dependencies` it's good idea, but beter if we add multiple keys like in original package (https://github.com/qpsolvers/qpsolvers/blob/65050939b6f32cf2e86c5bf9bcc3019427809d5c/pyproject.toml#L35-L49).

```suggestion
  passthru.optional-dependencies = let 
    clarabel_deps = [clarabel];
    ...
    open_source_solvers = [clarabel_deps cvxopt_deps ...];
   in  {
    clarabel = clarabel_deps,
    inherit open_source_solvers;
    ...
   };
```

It's free hand variant. I think it can be look better :)

---

i don't see the benefit for creating a key for each solver is when they can already be accessed by the package name.

---

But that's how `passthru.optional-dependencies` works.